### PR TITLE
 🐛  make sure the proxy values are set on the android pod

### PR DESCRIPTION
**Pull request by wei-lee**
This PR will:

1. Make sure the "-remoting" flag is set when running the Jenkins cli. This is required for running groovy scripts. It was the default mode before, but with the current version of Jenkins, this mode is deprecated and we need to explicitly add this flag to run the groovy scripts
2. Make sure the proxy env vars are set for the android pod template.

To verify:

Remote the android pod template from osm4. Run the playbook and it should recreate the pod template, with the correct env vars created.

ping @laurafitzgerald @PhilipGough @luigizuccarelli for review

### CODE REVIEW

**Comment by laurafitzgerald** [comment]
@wei-lee :eyes:

File: configure-buildfarm/templates/podtemplate-config.j2 [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.csanchez.jenkins.plugins.kubernetes.*
+import org.csanchez.jenkins.plugins.kubernetes.volumes.*
+import jenkins.model.*
+import java.util.ArrayList
+import java.util.List
+
+def NAME="android"
+def IMAGE="{{ android_slave_image }}"
+def ENVAR_NAME="ANDROID_HOME"
+def ENVAR_VALUE="/opt/android-sdk-linux"
```
Comment (laurafitzgerald) [inline configure-buildfarm/templates/podtemplate-config.j2]: Could we rename these ENUM names to ANDROID_HOME_NAME and ANDROID_HOME_VALUE or similar.

**Comment by laurafitzgerald** [comment]
@wei-lee added some new changes
**What**
- If the template exists
Check the existing env vars
Update them if they have changed
- If new template
- Create env vars as normal

Updated log statements used in the script. 
Now outputting this to the user.

**Why**
The plugin was duplicating (rather than recreating) the env vars when the playbook was run more than once.

**Verification**
_Step_
- Delete the pod template on osm4 (if present)
- Run:
```
ansible-playbook -i inventories/engineering/osm4/osm4-hosts playbooks/buildfarm.yml -e "digger_project_name=digger" --tags=configure-buildfarm --vault-password-file ~/.osm4pass.txt 
```
_Expected Result_
- New pod template created with the correct entries.

_Steps_
- Change one of the env vars (proxy_url or proxy_gradle_opts) in the osm4 inventory file.
- Re-run the command.

_Expected Result_
- The changed env var should be updated
- No duplicate env vars should be present

**Comment by wei-lee** [comment]
@laurafitzgerald very good catch. I didn't realise the env vars will be duplicated as the code was recreating the env var list every time when it runs. It turns out that the `setEnVars` method is adding the new env vars to the existing list, rather than reset the envVars list completely (See https://github.com/carlossg/jenkins-kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java#L308).

I improved the format of the code. Verified it on OSM4.

**Pull request merged**